tests/framework/scheduler.py

```diff
@@ -25,20 +40,134 @@
 import os
 import re
 import sys
+import threading
+from pathlib import Path
 from random import random
 from select import select
 from time import sleep
 import pytest
-# Needed to force deselect nonci tests.
-from _pytest.mark import Expression, MarkMatcher
+
 from _pytest.main import ExitCode
+import host_tools.proc as proc
 
 from . import mpsing  # pylint: disable=relative-beyond-top-level
+from . import defs  # pylint: disable=relative-beyond-top-level
+
+
+class Suite:
+    """Declare a test suite given a name and list of paths."""
+
+    def __init__(self, name, path_list):
+        """Initialize using name and list of paths."""
+        self.name = name
+        self.path_list = [Path(path) for path in path_list]
+        self.ignored = []
+
+    def ignore_path(self, path):
+        """Ignore tests in this path."""
+        self.ignored.append(path)
+
+    def get_runnable_items(self, session_items):
+        """Return a list of runnable items given the session items."""
+        items = list(session_items)
+
+        # A test can match any of the patterns defined by the batch,
+        # in order to get assigned to it.
+        to_remove = []
+        for item in items:
+            # Check if the file path matches any of the path lists
+            matches_batch = re.search(
```

Comment:
This method could be split into two: `matches_path()` and `is_ignored_path()` which would make it easier to read